routes/w3s.ts

```diff
@@ -28,17 +28,53 @@ async function parseProof(data) {
 export default defineEventHandler(async (event) => {
   const requestObject = event.node.req
 
-  const body = await watchData(requestObject)
+  const storage = useStorage('redis')
+
+  // Extract and validate the authToken
+  let authToken = requestObject.headers['authorization']
+
+  if (Array.isArray(authToken)) {
+    authToken = authToken[0] // Take the first token if there are multiple
+  }
 
+  authToken = authToken?.split(' ')[1]
+  if (!authToken) {
+    return { error: 'No authentication token provided' }
+  }
+
+  let inStorage = await storage.hasItem(authToken)
+  let tokenData
+
+  // this logic could be better. we should check if its expired, if so revalidate etc.
```

Comment:
It looks like you did indeed address this? Is that the case?

Reply:
oh yeah, old comment. can remove